internal/handlers/staking_restrictions_hooks.go

```diff
@@ -0,0 +1,141 @@
+package handlers
+
+import (
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Assert proper implementation of StakingRestrictions
+var _ stakingtypes.StakingHooks = StakingRestrictionHooks{}
+
+const (
+	// The default concentration of bonded tokens a validator is allowed as a multiple of equal shares
+	DefaultConcentrationMultiple = 5.5 // Amounts to ~8% with 68 validators
+	// Maximum Allowed Cap for Bonded stake of any single validator
+	DefaultMaxCapPercent = 0.33
+	// Minimum Allowed Cap for Bonded stake of any single validator
+	DefaultMinCapPercent = 0.05
+)
+
+type RestrictionOptions struct {
+	MaxConcentrationMultiple float32
+	MaxBondedCapPercent      float32
+	MinBondedCapPercent      float32
+}
+
+// DefaultRestictionOptions are default constraints that prevent single point of failure on validators
+var DefaultRestictionOptions = &RestrictionOptions{
+	MaxConcentrationMultiple: DefaultConcentrationMultiple,
+	MaxBondedCapPercent:      DefaultMaxCapPercent,
+	MinBondedCapPercent:      DefaultMinCapPercent,
+}
+
+// UnlimitedRestrictionOptions are used to remove restrictions for validator staking limits from delegations
+var UnlimitedRestrictionOptions = &RestrictionOptions{
```

Comment:
This isn't currently being used anywhere. Is this for later use?

Reply:
Its a demonstration configuration for the appropriate way to remove restrictions.